oepcie/yuanphil_impl/oe_ctx.h

```diff
@@ -0,0 +1,31 @@
+#ifndef _INCL_GUARD_ONE
+#define _INCL_GUARD_ONE
+
+#include "oe_ctx_impl.h"
+
+typedef struct oe_ctx_impl *oe_ctx;
+
+int oe_init(oe_ctx* ctx);
+
+oe_ctx* oe_create();
+
+void oe_destroy(oe_ctx* ctx);
+
+int oe_set_opt(oe_ctx* state, int option, const char* value);
+
+/*
+ * For functional purposes, here we are going to assume that value points to
+ * NULL, and we will allocate the string to return. It is up to the use to cl-
+ * ean up memory.
+ */
+int oe_get_opt(oe_ctx* state, int option, char* value);
+
+int oe_write_reg(oe_ctx* state, int device_id, int addr, int value);
```

Comment:
We will look into the const, but just looking at it I think const will be okay. State is a **oe_ctx_impl, so we wont actually be modifying the address stored in state, just the structure that it points to. I'm recalling some semantic about "const type*" vs "type* const".. The difference between a constant pointer and a pointer to const. I think if we use the latter then it should be fine. Will change the idx argument stuff as well.